doc/ref_model/chapters/appendix-a.md

```diff
@@ -29,13 +29,21 @@ The goal is not to be prescriptive on how to re-architect existing or architect
 ## A.4 Links
 
 <a name="A.5"></a>
-## A.5 Hardware-Dependent Coding Policies
+## A.5 CNTT Policies for Managing Non-Conforming Technology and Transition
 
-As described in the Principles sections of RM Chapter 1, features that require hardware-dependent code in the workload are prohibited in CNTT compliant implementations. This principle is henceforth referred to as the "Abstraction Principle". Note, this is not the case for the Compute node host software (e.g., host OS). Within the Infra, hosts are expected to have software that is customized for the specific hardware equipped. However, the intent is that these software drivers and higher layers will abstract the Capabilities they enable, thereby exposing them with an open API. An example of exposing capabilities in this manner is implemented in the Virtio family of APIs. This requirement is in support of VNF abstraction and portability of VNFs across the Infra landscape.
+There are multiple situations where a policy, comprised of one or more compromises and/or transitions is required to address technology that does not presently conform to CNTT mandates or strategy, and hence requires explicit direction to prescribe how the situation will be treated in the present, as well as in the future. This informs application designers how RC and ultimately OVP will react when encountering such technologies during the qualification process, including flagging warnings and potentially errors which could prevent issuance of an OVP badge. 
 
-CNTT realizes there are implications to this and limitations to the ability to live by the Abstraction Principle. A textbook example of a Capability that transgresses this principle, is SR-IOV. Other, less notable, yet very important examples include vGPUs and other acceleration hardware, such as FPGA. A less obvious, yet critically important example is the VNF program(s) itself. As workloads execute native microarchitecture opcodes, those instruction sets effectively constitute an ABI (Application Binary Interface). Additionally, these programs may or may not, attempt to execute vendor-specific extensions to standard instruction sets, such as x86 or ARM.
+<a name="A.5.1"></a>
+### A.5.1 Feature Availability
+One such case is where CNTT strategically deems a given capability as mandatory, but the feature is a roadmap item, under development or otherwise unavailable. To address this scenario, a policy can be created to recognize the current state of the technology, and identify a Time Point (TP) in VNF Evolution when the feature will become mandatory for RC purposes. 
```

Comment:
I would like to discuss whether we introduce the concept of 'time point' or use an alternate mechanism.  Examples of these alternates are:
- defining a new generation of workload profiles, where the capabilities are explicitly indicated or contra-indicated
- defining a new version of the Reference Model document

It is likely that the changes required to migrate from a non-conforming technology to a conforming technology will require re-configuration, up to and including, hardware changes on the host.  This is something that would have to be rolled out gradually across a NFV.  VNFs would find it very useful to have a mechanism to place their workloads on the newly compliant infrastructures, or to steer them towards the older infrastructure until such time as the workload can be revised to meet the new compliance requirement.

Reply:
@kedmison Hi Kelvin, 

>
> I would like to discuss whether we introduce the concept of 'time point'

This PR/section is just a consumer of the VNF Evolution Framework, and really isn't intended to describe it. Therefore, whatever the VNF Ev metric ends up as, be it TPs, tide cycles, solar events, or whatnot, it will be inherited here. Please refer to Issue #750 for the VNF Evolution creation topic.

>
> VNFs would find it very useful to have a mechanism to place their workloads on the newly compliant infrastructures, or to steer them towards the older infrastructure...

Is there a limitation you've got in mind, that would prevent the existing H/W Profiles from being able to accommodate the differentiation you're looking for?

>
> some thought on whether to, and/or how to separate the discussion of SR-IOV policy from this framework.

As mentioned above, if I have to, I'll just peel the compromises off, and place them in a separate PR. So, how to, is not a lot of effort logistically. Whether to, will just be based on whether the discussion will delay merging the framework.

Reply:
hi @markshostak 
> Is there a limitation you've got in mind, that would prevent the existing H/W Profiles from being able to accommodate the differentiation you're looking for?

One example: in some cases, I can envision that an operator may want to operate an environment where VNFs are leveraging deprecated mechanisms at the same time as other VNFs are leveraging the new/replacement mechanisms. 

While a workload leveraging the newer capability may be able to operate on the newer host configuration, it cannot work on the older one.  And, it's possible that the inverse is true too.  Operators thus need to support a heterogeneous set of hosts, consisting of deprecated-mechanism-enabled hosts and and replacement-mechanism-enabled hosts, in their equipment fleets.   

I thus anticipate that the use of a workload profile generation, which we currently do not have, would be a very useful abstraction in managing placement of workloads on compute nodes of the correct configuration.

Reply:
@kedmison Don't the exiting h/w profiles allow us to do that? Thx, -Mark

Reply:
At present, I don't believe so.  The naming convention in chapter 4.2.7 has no generational aspect to the naming, giving us no way to version a basic gen1 from a basic gen2.

Reply:
I'll add comments to #750 as was discussed in the Technical Committee meeting.